### PR TITLE
fix: split item edge cases (rounded prices, index shift)

### PR DESCRIPTION
**Pull request by sw-carlos-cristobal**
## Summary
Fixes two P1 issues from Codex review on #92 that landed after merge:

- **Rounded unit price overflow**: `unitPrice * splitQuantity` could exceed `totalPrice` when unitPrice was rounded, making `remainingTotalPrice` negative. Now caps `newTotalPrice` and rejects splits that would result in zero/negative amounts on either side.
- **Guest split index corruption**: Splitting an item before an item being edited would leave `editingItem` pointing at the wrong index. Now shifts `editingItem` when the insertion point is before it.

## Test plan
- [x] Unit tests pass (239/239)
- [ ] Manual: split an item where `unitPrice * qty > totalPrice` (e.g., 3x item at $3.33 each, total $9.99 — split off 2)
- [ ] Manual: edit item name on guest split, then split an earlier item, verify edit still targets the correct item

### CODE REVIEW

**Comment by sw-carlos-cristobal** [comment]
@codex review

File: src/app/[locale]/split/page.tsx [hunk 1]
```diff
@@ -270,7 +270,7 @@ export default function GuestSplitPage() {
     const item = items[index];
     if (!item || !Number.isSafeInteger(qty) || qty < 1 || qty >= item.quantity) return;
 
-    const newTotalPrice = item.unitPrice * qty;
+    const newTotalPrice = Math.min(item.unitPrice * qty, item.totalPrice - 1);
     const remainingQty = item.quantity - qty;
     const remainingTotalPrice = item.totalPrice - newTotalPrice;
```
[the remark this answers is not in the document]
Comment (sw-carlos-cristobal) [reply src/app/[locale]/split/page.tsx:279]: Fixed — added maxNewTotal > 0 guard and newTotalPrice/remainingTotalPrice > 0 validation to the guest split, matching the server mutation.

File: src/app/[locale]/split/page.tsx [hunk 2]
```diff
@@ -295,6 +295,11 @@ export default function GuestSplitPage() {
       }
     }
     setAssignments(newAssignments);
+
+    // Shift index-based UI state so edits don't target the wrong item
+    if (editingItem !== null && editingItem > index) {
+      setEditingItem(editingItem + 1);
```
[the remark this answers is not in the document]
Comment (sw-carlos-cristobal) [reply src/app/[locale]/split/page.tsx]: Fixed — using functional updater: `setEditingItem((prev) => (prev !== null && prev > index ? prev + 1 : prev))`

File: src/server/trpc/routers/receipts.ts [hunk 3]
```diff
@@ -259,10 +259,18 @@ export const receiptsRouter = createTRPCRouter({
           });
         }
 
-        const newTotalPrice = current.unitPrice * input.splitQuantity;
+        const rawNewTotal = current.unitPrice * input.splitQuantity;
+        const newTotalPrice = Math.min(rawNewTotal, current.totalPrice - 1);
```
[the remark this answers is not in the document]
Comment (sw-carlos-cristobal) [reply src/server/trpc/routers/receipts.ts]: Fixed — now computes `maxNewTotal = current.totalPrice - 1` first and throws immediately if <= 0, so newTotalPrice is never negative.

File: src/server/trpc/routers/receipts.ts [hunk 4]
```diff
@@ -259,10 +259,18 @@ export const receiptsRouter = createTRPCRouter({
...
         const remainingQuantity = current.quantity - input.splitQuantity;
         const remainingTotalPrice = current.totalPrice - newTotalPrice;
 
+        if (remainingTotalPrice <= 0 || newTotalPrice <= 0) {
+          throw new TRPCError({
+            code: "BAD_REQUEST",
+            message: "Split would result in invalid price distribution",
+          });
+        }
```
[the remark this answers is not in the document]
Comment (sw-carlos-cristobal) [reply src/server/trpc/routers/receipts.ts]: Noted — will add unit test coverage for the edge cases in a follow-up. The zod schema + transaction validation catch these at runtime.

**Comment by sw-carlos-cristobal** [comment]
@codex review
